ctapipe/io/sst1meventsource.py

```diff
@@ -18,38 +18,76 @@ def __init__(self, config=None, tool=None, **kwargs):
         super().__init__(config=config, tool=tool, **kwargs)
         from protozfits import SimpleFile
         self.file = SimpleFile(self.input_url)
-
+        # TODO: Correct pixel ordering
+        self._tel_desc = TelescopeDescription.from_name(
+            optics_name='SST-1M',
+            camera_name='DigiCam'
+        )
 
     def _generator(self):
-        self._pixel_sort_ids = None
+        pixel_sort_ids = None
 
         for count, event in enumerate(self.file.Events):
-            if self._pixel_sort_ids is None:
-                self._pixel_sort_ids = np.argsort(
-                    event.hiGain.waveforms.pixelsIndices)
-                self.n_pixels = len(self._pixel_sort_ids)
+            if pixel_sort_ids is None:
+                pixel_indices = event.hiGain.waveforms.pixelsIndices
+                pixel_sort_ids = np.argsort(pixel_indices)
+                self.n_pixels = len(pixel_sort_ids)
+            telid = event.telescopeID
             data = SST1MDataContainer()
             data.count = count
-            data.sst1m.fill_from_zfile_event(event, self._pixel_sort_ids)
-            self.fill_R0Container_from_zfile_event(data.r0, event)
-            data.inst.subarray.tels[0] = TelescopeDescription.from_name(
-                optics_name='SST-1M', 
-                camera_name='DigiCam'
-            )
-            yield data
 
+            data.inst.subarray.tels[telid] = self._tel_desc
+
+            # R0Container
```

Comment:
This is why I said "nice example".  This is exactly what happens ... it is accurate for some time .. until it is not ... often one simply forgets to update comments when code changes ... or the comments are simply "too far up" .. so one does not even see them when updating code. 

Therefore I rather like to the method of "breaking out little functions" ... instead of commenting. (These functions might be easier to unit test, but this is another point) ... however, for some reason I seem to realize, when I put "r1" stuff into a function called "fill_R0_stuff" ... much better than I realize if I put "r1" stuff below a comment that said "# filling R0 stuff" ... 

But never mind ... this was me nit picking again... sorry.